pkg/pillar/utils/deserialize.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2017-2023 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"io"
+	"os"
+	"reflect"
+)
+
+func readFile(filename string) ([]byte, error) {
+	// Check if the file exists
+	if _, err := os.Stat(filename); os.IsNotExist(err) {
+		log.Errorf("File %s does not exist", filename)
+		return nil, err
+	}
+
+	// Open the file
+	file, err := os.Open(filename)
+	if err != nil {
+		log.Errorf("Failed to open file %s, %s", filename, err.Error())
+		return nil, err
+	}
+	defer file.Close()
+
+	// Read the data
+	data, err := io.ReadAll(file)
+	if err != nil {
+		log.Errorf("Failed to read data from file %s, %s", filename, err.Error())
+		return nil, err
+	}
+
+	return data, nil
+}
+
+func unmarshalJSON(data []byte, v interface{}) error {
```

Comment:
What is the point of this function, seems it just forwards the call to `json.Unmarshal`.

Reply:
Done